src/hotspot/share/memory/heapShared.cpp

```diff
@@ -398,8 +398,6 @@ void KlassSubGraphInfo::add_subgraph_object_klass(Klass* orig_k, Klass *relocate
       new(ResourceObj::C_HEAP, mtClass) GrowableArray<Klass*>(50, mtClass);
   }
 
-  assert(relocated_k->is_shared(), "must be a shared class");
```

Comment:
Why is this assert removed?

Reply:
The delete is reasonable since line 293 already assert that relocated_k is a relocated klass in shared region which must be shared class I think.

Reply:
`Klass::set_is_shared()` is called in `Klass::remove_unshareable_info()`. This PR has moved the calls to `remove_unshareable_info()` to a later stage. So when this assert is executed, `relocated_k->is_shared()` is false.

In the new commit [03f7485](https://github.com/openjdk/jdk/pull/227/commits/03f748555f304d4cc4d180ab989617bcebe508fb) I've restored the assert and changed it to the following:

```
assert(ArchiveBuilder::singleton()->is_in_buffer_space(relocated_k), "must be a shared class");
```